server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -362,45 +356,37 @@ public void parse(FieldMapper mainField, ParseContext context) throws IOExceptio
             context = context.createMultiFieldContext();
 
             context.path().add(mainField.simpleName());
-            for (ObjectCursor<FieldMapper> cursor : mappers.values()) {
-                cursor.value.parse(context);
+            for (FieldMapper mapper : mappers.values()) {
+                mapper.parse(context);
             }
             context.path().remove();
         }
 
         public MultiFields merge(MultiFields mergeWith) {
-            ImmutableOpenMap.Builder<String, FieldMapper> newMappersBuilder = ImmutableOpenMap.builder(mappers);
-
-            for (ObjectCursor<FieldMapper> cursor : mergeWith.mappers.values()) {
-                FieldMapper mergeWithMapper = cursor.value;
-                FieldMapper mergeIntoMapper = mappers.get(mergeWithMapper.simpleName());
+            Map<String, FieldMapper> newMappers = new HashMap<>();
+            for (FieldMapper mapper : mergeWith.mappers.values()) {
+                FieldMapper mergeIntoMapper = mappers.get(mapper.simpleName());
                 if (mergeIntoMapper == null) {
-                    newMappersBuilder.put(mergeWithMapper.simpleName(), mergeWithMapper);
+                    newMappers.put(mapper.simpleName(), mapper);
                 } else {
-                    FieldMapper merged = mergeIntoMapper.merge(mergeWithMapper);
-                    newMappersBuilder.put(merged.simpleName(), merged); // override previous definition
+                    FieldMapper merged = mergeIntoMapper.merge(mapper);
+                    newMappers.put(merged.simpleName(), merged); // override previous definition
                 }
             }
-
-            ImmutableOpenMap<String, FieldMapper> mappers = newMappersBuilder.build();
-            return new MultiFields(mappers);
+            return new MultiFields(newMappers);
         }
 
         @Override
-        public Iterator<Mapper> iterator() {
-            return StreamSupport.stream(mappers.values().spliterator(), false).map((p) -> (Mapper)p.value).iterator();
+        public Iterator<FieldMapper> iterator() {
+            return mappers.values().iterator();
         }
 
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             if (!mappers.isEmpty()) {
                 // sort the mappers so we get consistent serialization format
-                Mapper[] sortedMappers = mappers.values().toArray(Mapper.class);
-                Arrays.sort(sortedMappers, new Comparator<Mapper>() {
-                    @Override
-                    public int compare(Mapper o1, Mapper o2) {
-                        return o1.name().compareTo(o2.name());
-                    }
-                });
+                List<Mapper> sortedMappers = mappers.values().stream()
```

Comment:
I'm so happy we sorted this. I know we did it for an odd reason around loops in mapping updates, but it just makes the output nicer to read.

Reply:
I wonder, not that it matters, but, is it faster to do `list = new List(mappers.values()); Collections.sort(list);`. Sorting a stream feels funny to me because it is a forwards only data structure and feels like it'll amount to just sorting anyway. Whatever. It super doesn't matter.

Reply:
Oh! It's worth triple checking that this doesn't change the order from before. It sure looks the same to me but please make sure we have a test. We need the same order as before or things can get "fun" in a mixed version cluster.